src/app/confirm-dialog/confirm-dialog.component.ts

```diff
@@ -0,0 +1,67 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+import { Component, Inject, OnInit } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { environment } from 'src/environments/environment';
+import { CustomerDetailComponent } from '../customer-detail/customer-detail.component';
+import { Customer } from '../customers/customer';
+
+
+@Component({
+  selector: 'app-confirm-dialog',
+  templateUrl: './confirm-dialog.component.html',
+  styleUrls: ['./confirm-dialog.component.css']
+})
+export class ConfirmDialogComponent implements OnInit {
+  
+  name:string;
+  id:number;
+  customers!:Customer[];
+  
+
+  constructor(
+    private dialRef: MatDialogRef<CustomerDetailComponent>,
+    private httpClient:HttpClient,
+    @Inject(MAT_DIALOG_DATA) public data: any) {
+      this.name=data.name;
+      this.id=data.id;
+    }
+  
+
+  ngOnInit(): void {
+    this.getCustomers();
+  }
+  getCustomers()
+  {
+    this.httpClient.get<any>(environment.baseUrl+'customers').subscribe(
+    response=>{
+      console.log(response);
+      this.customers=response;
+    }
+    );
+  }
```

Comment:
When u pass the name and ID for the dialog, you don't really need to store this information and make an unnecessary API call. You can later on use the same ID to delete the customer.